torch/_inductor/sizevars.py

```diff
@@ -373,6 +374,9 @@ def symbolic_hint(self, expr: Expr) -> Expr:
 
     def size_hint(self, expr: Expr) -> int:
         out = self.symbolic_hint(expr)
+        # if `config.capture_scalar_outputs`` is true, allow unbacked symint in size hint
+        if torch._dynamo.config.capture_scalar_outputs and isinstance(out, sympy.Expr):
+            return out
```

Comment:
Shouldn't the relevant `size_hint` calls be updated to call `symbolic_hint` directly? Most calls to `size_hint` I'm aware of are explicitly because sympy expressions don't work in the code that comes afterwards.